docs/BUILDING.md

```diff
@@ -122,6 +123,63 @@ If you want to run the quick-lint-js program, you can find it in
 `build\Debug\quick-lint-js.exe`. Run it from a command prompt or PowerShell
 window.
 
+---
+
+### macOS and Linux: nix
+
+This is for advanced users only. You need to
+[install the Nix package manager](https://nixos.org/download.html) if you have
+not already done so.
+
+[Nix][] uses derivation files that contain instructions for automation on how
+to build and package software. The `dist/` folder contains such derivations for
+quick-lint-js and can be used to download dependencies and build it.
+
+#### 1. Download build dependencies
```

Comment:
I added a step 0 for nix. Not adding a separate step for `nix-shell` and the configure phase would have made the configure phase very long. I also didn't want to go and change all other build instructions since it seems to be a bit out of scope for this PR.